rero_ils/modules/documents/views.py

```diff
@@ -637,3 +575,38 @@ def get_articles(record):
             'pid':hit.pid
         })
     return articles
+
+
+@blueprint.app_template_filter()
+def online_holdings(document_pid, viewcode='global'):
+    """Find holdings by document pid and viewcode.
+
+    : param document_pid: document pid
+    : param viewcode: symbol of organisation viewcode
+    : return: list of holdings
+    """
+    holdings = []
+    organisation = None
+    if viewcode != current_app.config.get('RERO_ILS_SEARCH_GLOBAL_VIEW_CODE'):
+        organisation = Organisation.get_record_by_viewcode(viewcode)
+    from ..holdings.api import HoldingsSearch
+    query = HoldingsSearch()\
+        .filter('term', document__pid=document_pid)
+    if organisation:
+        query = query.filter('term', organisation__pid=organisation.pid)
+    results = query.source(['library', 'electronic_location', 'notes']).scan()
+    holdings = []
+    for record in list(results):
+        library = Library.get_record_by_pid(record['library']['pid'])
```

Comment:
I'am not sure we can access the values in an attribute list like this normaly we have to write `record.library.pid`.

Reply:
done